### ISSUE TITLE
Document that Bytes.forFieldGroup uses the first byte to store the length, consuming a byte and limiting the length to 255 bytes

### ISSUE THREAD

**Issue by minborg** [OpenHFT/Chronicle-Bytes#254]
Reproduce:

```
    static final class MassiveFieldHolder {

        @FieldGroup("b")
        long a0, a1, a2, a3, a4, a5, a6, a7, a8, a9, aa, ab, ac, ad, ae, af;
          
    }
   
```

Create using Bytes.forFieldGroup and the `writeLimit()` is 127 and not 128 as expected. Also, is seams to be an offset error so when actually reading the last bytes, it is non-zero

**Comment by JerryShea** [comment]
Hi @minborg while you are working in this area can I add another thing to the TODO list? EnbeddedBytes should be EmbeddedBytes

**Comment by minborg** [comment]
> Hi @minborg while you are working in this area can I add another thing to the TODO list? EnbeddedBytes should be EmbeddedBytes

Will do. See https://github.com/OpenHFT/Chronicle-Bytes/issues/260

**Comment by peter-lawrey** [comment]
For a Bytes in an embedded object, the length must be stored in that object as well. The assumption is that the length is a u8 i.e. uses the first byte available so the limit is one byte less than the amount of space allocated.
When Bytes are normally serialzied the length is prepended to the contents, but this is not possible for trivially copyable data.

**Comment by peter-lawrey** [comment]
@minborg Can you update any documentation you think would help to highlight this?

**Comment by minborg** [comment]
> @minborg Can you update any documentation you think would help to highlight this?

See the description in the initial comment. Perhaps this is not an error but a lack of documentation?

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.23ea9](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.23ea9), [BOM-2.23ea61](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.23ea61)
